sdk/monitor/query/MIGRATION.md

```diff
@@ -0,0 +1,211 @@
+        },
+        nil)
+    if err != nil {
+        //TODO: handle error
+    }
+    if res.Error != nil {
+        //TODO: handle partial error
+    }
+}
+```
+
+### `azlogs`
+
+The logs code for `azlogs` and `azquery` are very similiar. 
+
+```go
+import (
+	"context"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/monitor/query/azlogs"
+)
+
+func main() {
+    // create the client
+    cred, err := azidentity.NewDefaultAzureCredential(nil)
+    if err != nil {
+        //TODO: handle error
+    }
+    client, err := azlogs.Client(cred, nil)
+    if err != nil {
+        //TODO: handle error
+    }
+
+    // execute the logs query
+    res, err := client.QueryWorkspace(
+        context.TODO(),
+        workspaceID,
+        azlogs.QueryBody{
+            Query:    to.Ptr("<kusto query>"), // example Kusto query
+            Timespan: to.Ptr(azlogs.NewTimeInterval(time.Date(2022, 12, 25, 0, 0, 0, 0, time.UTC), time.Date(2022, 12, 25, 12, 0, 0, 0, time.UTC))),
+        },
+        nil)
+    if err != nil {
+        //TODO: handle error
+    }
+    if res.Error != nil {
+        //TODO: handle partial error
+    }
+}
+```
+
+## Query Metrics
+
+### `azquery`
+
+```go
+import (
+	"context"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/monitor/azquery"
+)
+
+func main() {
+    // create the metrics client
+    cred, err := azidentity.NewDefaultAzureCredential(nil)
+    if err != nil {
+        //TODO: handle error
+    }
+    client, err := azquery.NewMetricsClient(cred, nil)
+    if err != nil {
+        //TODO: handle error
+    }
+
+    // execute the metrics query
+    res, err := client.QueryResource(context.TODO(), "<resourceID>",
+        &azquery.MetricsClientQueryResourceOptions{
+            Timespan:        to.Ptr(azquery.NewTimeInterval(time.Date(2022, 12, 25, 0, 0, 0, 0, time.UTC), time.Date(2022, 12, 25, 12, 0, 0, 0, time.UTC))),
+            Interval:        to.Ptr("PT1M"),
+            MetricNames:     nil,
+            Aggregation:     to.SliceOfPtrs(azquery.AggregationTypeAverage, azquery.AggregationTypeCount),
+            Top:             to.Ptr[int32](3),
+            OrderBy:         to.Ptr("Average asc"),
+            Filter:          to.Ptr("BlobType eq '*'"),
+            ResultType:      nil,
+            MetricNamespace: to.Ptr("Microsoft.Storage/storageAccounts/blobServices"),
+        })
+    if err != nil {
+        //TODO: handle error
+    }
+}
+```
+
+### `armmonitor`
+
+The code in `armmonitor` is closer to the REST API; therefore, there are some name changes. Additionally, `azquery.MetricsClient` is separated into [different clients](#name-changes).
+
+```go
+import (
+    "context"
```

Comment:
Indentation on this line looks off

Reply:
fixed with [c79390d](https://github.com/Azure/azure-sdk-for-go/pull/22753/commits/c79390ddba46a599c6717d0bde75b9d3ccb46996)